### PR TITLE
Options not being sent on request

### PR DESCRIPTION
**Pull request by jose-rivas-lineten**

### CODE REVIEW

File: src/CurlRequest.php [hunk 1]
```diff
@@ -25,9 +25,8 @@ class CurlRequest
      */
     public function with(callable $callback)
     {
-        $clone = clone $this;
```
Comment (timohuovinen) [inline src/CurlRequest.php:28]: This removes immutability, which is required to keep consistent with how PSR handles the HTTP request and response objects (and also Slim)

File: composer.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
         }
     ],
     "require": {
-        "php": ">=5.3.0",
```
Comment (timohuovinen) [inline composer.json:25]: Version should be 5.4 to support square brackets arrays [].
Version numbers are ranges that the code supports, if we introduce new code that is only working in php 7.2, then we will update it to 7.2

**Comment by timohuovinen** [comment]
The reason why the options are not applied because one must do

```
$req = $req->with();
return $req;
```

Instead of

```
$req->with();
return $req;
```

because the `with` function keeps the instance immutable.

**Comment by jose-rivas-lineten** [comment]
because the immutability, the code should be different

**Comment by timohuovinen** [comment]
@jose-rivas-lineten How should it be different?

**Pull request closed without merging**